hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java

```diff
@@ -16,93 +16,87 @@
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotSame;
-import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertSame;
-import static org.junit.Assert.fail;
 
-@RunWith(HazelcastParallelClassRunner.class)
+@RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class PlanCacheIntegrationTest extends PlanCacheTestSupport {
+    private String mapName;
 
-    private final TestHazelcastFactory factory = new TestHazelcastFactory();
+    @BeforeClass
+    public static void setUp() {
+        initialize(2, smallInstanceConfig());
+    }
 
-    @After
-    public void after() {
-        factory.shutdownAll();
+    @Before
+    public void before() {
+        mapName = SqlTestSupport.randomName();
     }
 
     @Test
     public void testPlanIsCached() {
-        HazelcastInstance member = factory.newHazelcastInstance();
-        IMap<Integer, Integer> map = member.getMap("map");
+        IMap<Integer, Integer> map = instance().getMap(mapName);
         map.put(1, 1);
+        instance().getSql().execute("SELECT * FROM " + mapName);
 
-        PlanCache planCache = getPlanCache(member);
+        PlanCache planCache = getPlanCache(instance());
 
-        Plan plan = getPlan(member, "SELECT * FROM map");
         assertEquals(1, planCache.size());
-        assertSame(plan, planCache.get(plan.getPlanKey()));
+        SqlPlan prevPlan = planCache.get(planCache.getPlans().keys().nextElement());
 
-        Plan plan2 = getPlan(member, "SELECT * FROM map");
         assertEquals(1, planCache.size());
-        assertSame(plan, planCache.get(plan.getPlanKey()));
-        assertSame(plan, plan2);
+        SqlPlan actualPlan = planCache.get(planCache.getPlans().keys().nextElement());
+        assertSame(prevPlan, actualPlan);
     }
 
     @Test
     public void testPlanInvalidatedOnIndexAdd() {
-        HazelcastInstance member = factory.newHazelcastInstance();
-        IMap<Integer, Integer> map = member.getMap("map");
+        IMap<Integer, Integer> map = instance().getMap(mapName);
         map.put(1, 1);
 
-        PlanCache planCache = getPlanCache(member);
+        PlanCache planCache = getPlanCache(instance());
 
-        Plan plan = getPlan(member, "SELECT * FROM map");
+        instance().getSql().execute("SELECT * FROM " + mapName);
         assertEquals(1, planCache.size());
-        assertSame(plan, planCache.get(plan.getPlanKey()));
-
-        map.addIndex(IndexType.SORTED, "this");
+        PlanKey planKey = planCache.getPlans().keys().nextElement();
+        SqlPlan prevPlan = planCache.get(planKey);
 
         assertTrueEventually(() -> {
-            Plan plan2 = getPlan(member, "SELECT * FROM map");
+            map.put(2, 2);
+            map.addIndex(IndexType.HASH, "this");
```

Comment:
Why entry and index have to be continually added?

Reply:
Nothing wrong happened, but test was improved to more clear state :)